src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -388,6 +388,9 @@ public void writeGenericValue(@Nullable Object value) throws IOException {
         } else if (type == double[].class) {
             writeByte((byte) 20);
             writeDoubleArray((double[]) value);
+        } else if (type == BytesRef.class) {
+            writeByte((byte) 21);
+            writeBytesRef((BytesRef) value);
```

Comment:
if you rebase you'll get this change that was made upstream and merged back into our branch